main/src/ca/uwaterloo/flix/tools/pkg/FlixPackageManager.scala

```diff
@@ -17,25 +17,43 @@ package ca.uwaterloo.flix.tools.pkg
 
 import ca.uwaterloo.flix.api.Bootstrap
 import ca.uwaterloo.flix.tools.pkg.Dependency.FlixDependency
+import ca.uwaterloo.flix.tools.pkg.FlixPackageManager.downloadFromGitHub
 import ca.uwaterloo.flix.tools.pkg.github.GitHub
-import ca.uwaterloo.flix.util.Result.{Err, Ok, ToOk}
+import ca.uwaterloo.flix.util.Result.{Err, Ok, ToOk, traverse}
 import ca.uwaterloo.flix.util.Result
+import org.apache.commons.io.FileUtils
 
 import java.io.{IOException, PrintStream}
 import java.nio.file.{Files, Path, StandardCopyOption}
 import scala.util.Using
 
 object FlixPackageManager {
 
+  //All Maven dependencies found transitively from Flix dependencies
+  val mavenSet = collection.mutable.Set.empty[String]
+
+  //The total set of Flix dependencies found
+  private val flixSet = collection.mutable.Set.empty[FlixDependency]
+
+  //The set of Flix dependencies where transitive dependencies have not yet been found
+  //TODO: how to make this prettier?
+  private var workSet = collection.mutable.Set.empty[FlixDependency]
```

Comment:
I also need to have mutable sets of Flix dependencies that are being passed around between functions (specifically findTransitiveDependencies() and processTomlFile()). I made these into fields like the set of Maven dependencies, but these fields are private. This makes it easy to pass the sets around, but is that ok or would you prefer something different? If they should not be fields, I will probably need to inline processTomlFile() to make it work.

Reply:
Unfortunately again we can never use mutable state due to the existence of multiple Flix compiler instances within the same JVM.

Reply:
Okay I will do it without mutable state. But why is that not a problem in Bootstrap then?

Reply:
There can be multiple Bootstrap classes right? You can just create a new Bootstrap object if you want.